pcap-mode.el

```diff
@@ -362,6 +374,32 @@ Argument FILTER-VALUE corresponds to the exact set of filters passed to `pcap-mo
   (if pcap-mode-reload-pcap-when-filter-changes
       (pcap-mode-reload-file)))
 
+(defun pcap-mode-set-named-filter (filter-name)
+  "Choose a predefined filter and apply it.
+Filters are obtained from the `pcap-mode-dfilters-file'."
+  (interactive (list (ido-completing-read
+		      "Filter name: "
+		      (if (file-readable-p pcap-mode-dfilters-file)
+			  (with-temp-buffer
+			    (insert-file-contents pcap-mode-dfilters-file)
+			    (goto-char (point-min))
+			    (cl-loop while (re-search-forward "^\"\\([^\"]*\\)\"" nil t)
+				     collect (match-string 1)))
+			(error "Can't read dfilters file: %s" pcap-mode-dfilters-file)))))
+  (let (filter)
+    (if (file-readable-p pcap-mode-dfilters-file)
+	(progn (with-temp-buffer
+		 (insert-file-contents pcap-mode-dfilters-file)
+		 (goto-char (point-min))
+		 (search-forward filter-name)
+		 (setq filter (buffer-substring-no-properties (+ 2 (point)) (line-end-position))))
+	       (pcap-mode-set-tshark-filter (concat "'" filter "'")))
+      (error "Can't read dfilters file: %s" pcap-mode-dfilters-file))))
+
+(defun pcap-mode-clear-filter nil
```

Comment:
Can you split this series into three commits.  Commit one can fix the group binding for the custom variables above, commit two can add the clear-filter, commit three can add the new scanning code.

ALSO, this function has no documentation.